xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/internal/DefaultBeanDescriptor.java

```diff
@@ -222,6 +234,16 @@ protected void extractPropertyDescriptor(Field field, Object defaultInstance)
             // set parameter type
             desc.setPropertyType(field.getGenericType());
 
+            // set parameter display type
+            PropertyDisplayType displayTypeAnnotation = field.getAnnotation(PropertyDisplayType.class);
+            Type displayType;
+            if (displayTypeAnnotation != null) {
+                displayType = displayTypeAnnotation.value();
+            } else {
+                displayType = desc.getPropertyType();
+            }
+            desc.setDisplayType(displayType);
```

Comment:
There's a bit of code duplication here, not much, but I would still put it in a separate private method.